packages/engine-core/src/common/Engine.ts

```diff
@@ -25,16 +25,20 @@ export type BatchTransactionOptions = {
 
 export type InteractiveTransactionOptions<Payload> = Transaction.Info<Payload>
 
-export type RequestOptions<InteractiveTransactionPayload> = {
+export type GraphQLQuery = {
```

Comment:
Depending on what you do next, maybe a discriminant property will bee needed here

Reply:
Haven't decided on that yet. But if it will be necessary, I'll add it along with JSON protocol.